XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,90 @@
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways. Therefore an app may not know how to correctly render in its UI `conversationId` and `metadata` for conversations originating from other apps. We propose a schema to standardize the use of conversation `metadata` for the purpose of app brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `displayInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: "mydomain.xyz/abc/qrs",
+    metadata: { 
+        "displayInfo.prettyName": "My company", 
+        "displayInfo.profileImage": "mydomain.xyz/assets/myimage.png", 
+        "displayInfo.primaryColor": "#ffffff" }
+}
+```
+
+Example `displayInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: "galaxy.chat/dm/uniqueIdentifier",
+    metadata: { 
+        "displayInfo.prettyName": "Galaxy", 
+        "displayInfo.profileImage": "galaxychat.xyz/brandassets/logo.png", 
+        "displayInfo.primaryColor": "#6865B8" }
+}
+```
+
+The `profileImage` must meet the following criteria:
+
+- Aspect ratio: 1:1
+- Minimum resolution: 100x100
+- Maximum resolution: 800x800
+- Format: .PNG, .WEBP
+
+The `primaryColor` must be a hex color code.
+
+## Backward compatibility
+
+Existing conversations will not be affected by the adoption of the new schema; only conversations created after the new schema is finalized will follow the updated schema.
+
+To maintain backward compatibility, clients may consider rendering popular but older schemas (such as Lens), present in conversations created before the finalization of this XRC, in a manner consistent with their updated schema that follows this XRC.
+
+Old Lens schema:
+
+```json
+{
+        conversationId: "lens.dev/dm/${memberA}-${memberB}"
+        metadata: {}
+}
+```
+
+New Lens schema
+
+```json
+{
+    conversationId: "lens.dev/dm/${memberA}-${memberB}"
+    metadata: { 
+        "displayInfo.prettyName": "Lens", 
+        "displayInfo.profileImage": "lens.xyz/assets/myimage.png", 
+        "displayInfo.primaryColor": "#ffffff" }
+}
+```
+
+## Security considerations
+
+The `metadata` fields can be spoofed by malicious apps to display names and profile images that degrade the user experience and harm brand perception. A mechanism for apps to sign payloads such as conversation metadata and messages can enable frontends to verify the sending client's identity and prevent such spoofing. An affordance for client verifiability is under consideration.
```

Comment:
```suggestion
A brand's chosen metadata fields and content may be co-opted by other apps to display the same names and profile images which could impact the user experience and perception of a message's origin or affiliation. A mechanism for apps to sign payloads such as conversation metadata and messages could enable frontends to verify the sending client's metadata authenticity and prevent such spoofing. An affordance for verifying metadata could be considered in a future proposal.
```

I'm suggesting that we lighten the language on "spoofing" and "malicious" as it's simply an open space, and there's nothing currently preventing a project from co-opting some labels. Not ideal, but we should be careful about using such pointed language as it may diminish its value in other circumstances.